docs/protocol-and-architecture.md

```diff
@@ -0,0 +1,93 @@
+# Mediachain -- Protocol and Architecture
+
+In this document, we'll review the distributed architecture of Mediachain and
+describe the protocols and interfaces that connect the various pieces.
+
+Conceptually, Mediachain is broken into three core pieces: the **transactors**,
+responsible for maintaining global consensus regarding the
+[journal](data-structures.md#the-journal) (to be described); the **clients**
+that stream all operations (writes, updates, etc.) from the transactor quorum in
+order to process them and/or provide unique projections of the dataset; and the
+**storage layer** (at present DynamoDB, soon IPFS) responsible for housing the
+blockchains in a publicly accessible manner.
+
+## Transactors
+
+The transactor quorum is currently implemented as a
+[copycat](https://github.com/atomix/copycat) state machine. `copycat`
+features its own serialization library and RPC techniques, so Mediachain
+features a facade layer--implemented in [gRPC](http://grpc.io)--providing
+API support for multiple languages.
+
+### Facade
+
+The facade, written in Scala, communicates to the `copycat` quorum via
+`copycat`'s (JVM-only) API so end-users don't have to. Furthermore, the facade
+can interact with the **storage layer** to facilitate common requests:
+
+- **Replay/stream the blockchain from a given offset**: Clients that have either
+  disconnected from the **transactor quorum** or are connecting for the first
+  time can catch up to the current state of the blockchain. Once subscribed to
+  the facade, an application will receive a stream of all
```

Comment:
This is no longer true... subscribers now only get the current block + future updates, and need to follow the chain back to catchup.